### ISSUE TITLE
Problems with generated images

### ISSUE THREAD

**Issue by YangJae96** [chenhao2345/GCL#5]
Hi, I have a few problem with generating synthetic persons using your code in  examples/generate_data.py.!

This is the resuls what I get from original image to generated image.
![0002_c1s1_000451_03](https://user-images.githubusercontent.com/57695621/132230828-43d080a2-1c41-47d6-8c00-a0787b4ee10c.jpg)

![0002_c1s1_000451_03](https://user-images.githubusercontent.com/57695621/132230577-0c0745e1-7f1a-403f-96bc-e266b6853cf4.jpg)

As you can see, the result is not good..

I have resumed your model same as when ./train_stage3_market.sh starts.
Below is the model(generator, discrimminator) weights when  ./train_stage2_market.sh  ends. 
So, I resumed your model from those weights from the code below 

" 
 trainer = DGNet_Trainer(config, model_1, args.idnet_fix).cuda()
  iterations = trainer.resume(checkpoint_path, hyperparameters=config)
"

![image](https://user-images.githubusercontent.com/57695621/132231199-bad5eb97-3d50-4015-b23d-dbc8857622e5.png)

![image](https://user-images.githubusercontent.com/57695621/132231605-82080569-f517-40cf-ad21-7758e2dc2c20.png)

Actually, I trained your model with stage3,  -> ./train_stage3_market.sh 
and saved the model_best.pth in the outputs folder like below images.
![image](https://user-images.githubusercontent.com/57695621/132231686-f1cd9968-6a2a-439c-b990-4c854e44c199.png)
![image](https://user-images.githubusercontent.com/57695621/132231749-8070ac8e-89a3-435d-a805-78d57467a268.png)

So, I tried to bring your model_best.pth and resume it, hopefully thinking it will generate good results, but I could not resume your model because of the difference of the weights and models,.
![image](https://user-images.githubusercontent.com/57695621/132232050-a4d28a96-790b-40b9-80d0-fbb0ce17faf7.png)

What should I do to generate good results of your training datasets?? 
Thanks in advance!

**Comment by chenhao2345** [comment]
I added `trainer.module.save(checkpoint_directory, iterations)` in Line 319 of main.py so that you can store weights of stage 3.
 
You got the correct ReID results, but not generation results. It seems to me that something is wrong in generate_data.py. I will rerun the code to figure out the problem.

**Comment by chenhao2345** [comment]
Another question: Did you also get bad quality images in Tensorboard by running`tensorboard --logdir logs/market_init_JVTC_unsupervised/`
?

**Comment by YangJae96** [comment]
> Another question: Did you also get bad quality images in Tensorboard by running`tensorboard --logdir logs/market_init_JVTC_unsupervised/`
> ?

Actually I tried to use the tensorboard, but I was using ssh remote server and docker. I am now trying to make tensorboard work, but I'm still working on it. 
->  I could not see the results of your tensorboard yet. So, I was trying to save the results in the folder by using examples/generate_data.py!

 Actually I couldn't resume the model.best.pth weights(which is the Best Re-ID result)  into the model in examples/generate_data.py. 
 So, do I only need to load the the **best Re-ID result weights** of **id_net.pt**, **generator.pt** and **discrimminator.pt** by using your code ?
model_1 = create_model(args)
trainer = DGNet_Trainer(config, model_1, args.idnet_fix).cuda()
iterations = trainer.resume(checkpoint_path, hyperparameters=config)


> It seems to me that something is wrong in generate_data.py. I will rerun the code to figure out the problem. 
-> Thank you so much.

**Comment by YangJae96** [comment]
![imgs_nv](https://user-images.githubusercontent.com/57695621/132356890-4019c27e-f4be-4067-87bf-d6a77ff04d0d.jpg)

Actually, this is the image saved during training in stage3 when re-ID score is the best.  It seems okay. I think the problem in examples/generate_data.py, the meshes are not properly brought in with the original images.

**Comment by chenhao2345** [comment]
Yes, you are right. It was the problem of incoherent data-dir and mesh-dir format in Line 240 of generate_data.py. I just updated them. Please use the new generate_data.py. 

For reconstruction, use `python examples/generate_data.py --mode recon`
For new views, use `python examples/generate_data.py --mode nv --degree 45` (degree could be 45, 90, 135, 180, 225, 270, 315)

**Comment by YangJae96** [comment]
Thank you. It is now working!
